src/Core/maps/src/Handlers/Map/MapHandler.Android.cs

```diff
@@ -0,0 +1,632 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using Android;
+using Android.Content.PM;
+using Android.Gms.Maps;
+using Android.Gms.Maps.Model;
+using Android.OS;
+using AndroidX.Core.Content;
+using Java.Lang;
+using Java.Util.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Maui.Graphics;
+using Microsoft.Maui.Graphics.Platform;
+using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Platform;
+using ACircle = Android.Gms.Maps.Model.Circle;
+using APolygon = Android.Gms.Maps.Model.Polygon;
+using APolyline = Android.Gms.Maps.Model.Polyline;
+using Math = System.Math;
+
+namespace Microsoft.Maui.Maps.Handlers
+{
+	public partial class MapHandler : ViewHandler<IMap, MapView>
+	{
+		bool _init = true;
+
+		MapCallbackHandler? _mapReady;
+		MapSpan? _lastMoveToRegion;
+		List<Marker>? _markers;
+		IList? _pins;
+		IList? _elements;
+		List<APolyline>? _polylines;
+		List<APolygon>? _polygons;
+		List<ACircle>? _circles;
+
+		public GoogleMap? Map { get; set; }
+
+		static Bundle? s_bundle;
+
+		public static Bundle? Bundle
+		{
+			set { s_bundle = value; }
+		}
+
+		protected override void ConnectHandler(MapView platformView)
+		{
+			base.ConnectHandler(platformView);
+			_mapReady = new MapCallbackHandler(this);
+			platformView.GetMapAsync(_mapReady);
+			platformView.LayoutChange += MapViewLayoutChange;
+		}
+
+		protected override void DisconnectHandler(MapView platformView)
+		{
+			base.DisconnectHandler(platformView);
+			platformView.LayoutChange -= MapViewLayoutChange;
+
+			if (Map != null)
+			{
+				Map.SetOnCameraMoveListener(null);
+				Map.MarkerClick -= OnMarkerClick;
+				Map.InfoWindowClick -= OnInfoWindowClick;
+				Map.MapClick -= OnMapClick;
+			}
+
+			_mapReady = null;
+		}
+
+		protected override MapView CreatePlatformView()
+		{
+			MapView mapView = new MapView(Context);
+			mapView.OnCreate(s_bundle);
+			mapView.OnResume();
+			return mapView;
+		}
+
+		public static void MapMapType(IMapHandler handler, IMap map)
+		{
+			GoogleMap? googleMap = handler?.Map;
+			if (googleMap == null)
+				return;
+
+			googleMap.MapType = map.MapType switch
```

Comment:
It's not essential, but could we move this to extension methods like in other Handlers? This way, someone extending the Handler could easily access this logic.